lua/neogit/popups/tag/actions.lua

```diff
@@ -0,0 +1,161 @@
+local M = {}
+local git = require("neogit.lib.git")
+local client = require("neogit.client")
+local utils = require("neogit.lib.util")
+local FuzzyFinderBuffer = require("neogit.buffers.fuzzy_finder")
+
+--- Gets revisions
+---@return table
+local function get_revisions()
+  local revisions = git.cli["for-each-ref"].format('"%(refname:short)"').call():trim().stdout
+  for i, str in ipairs(revisions) do
+    revisions[i] = string.sub(str, 2, -2)
+  end
+  return revisions
+end
+
+--- Outputs a list of tags locally
+---@return table|nil List of tags.
+local function list_tags()
+  local revisions = git.cli.tag.args("-l").call():trim().stdout
+  if #revisions == 0 then
+    return nil
+  end
+  return revisions
+end
+
+function M.create_tag(popup)
+  local value = vim.fn.input { prompt = "Tag name: " }
+  if not value then
+    return
+  end
+  local options = get_revisions()
+  local selected_branch = FuzzyFinderBuffer.new(options):open_async()
+  if not selected_branch then
+    return
+  end
+  local args = popup:get_arguments()
+  if vim.tbl_count(args) > 0 and not vim.tbl_contains(args, "--annotate") then
+    table.insert(args, "--annotate")
+  end
+  client.wrap(git.cli.tag.arg_list({ value, selected_branch }).apply_flags(args), {
```

Comment:
Just to understand, why use both the new `apply_flags` and `arg_list`?

Reply:
Is there another way to apply flags dynamically? I need to do `git tag {list of tags} {list of user flags}`

Reply:
If we compare `arg_list` and `apply_flags`:
```lua
    if action == "arg_list" then
      return function(args)
        for _, v in ipairs(args) do
          table.insert(tbl[k_state].arguments, v)
        end
        return tbl
      end
    end

    if action == "apply_flags" then
      return function(flags)
        for _, v in ipairs(flags) do
          table.insert(tbl[k_state].options, v)
        end
        return tbl
      end
    end
```

And then where they are turned into a command:
```lua
  local function to_process(verbose, suppress_error, ignore_code)
    local cmd = {}

    for _, o in ipairs(state.options) do
      table.insert(cmd, o)
    end

    for _, arg in ipairs(state.arguments) do
      if arg ~= "" then
        table.insert(cmd, arg)
      end
    end

    ...
  end
```

Options just get inserted before arguments, so if you do like:
```lua
git.cli.tag.arg_list(util.merge(args, { value, selected_branch }))
```

That should, unless I'm mistaken, produce the same result.